### PR TITLE
Update android_in_app_purchases.rst

### PR DESCRIPTION
**Pull request by avnerh1**

### CODE REVIEW

**Comment by avnerh1** [comment]
- Added an example of using ``_on_purchases_updated``, which is the right way to handle the result of `payment.purchase()`
- Added check for `purchase.purchase_state`
- Added some info about consuming a product and about using `queryPurchases()`

**Comment by troy-lamerton** [comment]
Looks good to me. Checking the purchase state is essential which is the main thing added.

**Comment by m4gr3d** [comment]
cc @timoschwarzer

File: tutorials/platform/android/android_in_app_purchases.rst [hunk 1]
```diff
@@ -119,10 +132,11 @@ Full example:
     var query = payment.queryPurchases("inapp") # Or "subs" for subscriptions
     if query.status == OK:
         for purchase in query.purchases:
-            if purchase.sku == "my_iap_item":
-                premium = true # Entitle the user to the content they bought
+            if purchase.sku == "my_iap_item" and purchase.purchase_state==1:
```
Comment (timoschwarzer) [inline tutorials/platform/android/android_in_app_purchases.rst:135]:
```suggestion
            if purchase.sku == "my_iap_item" and purchase.purchase_state == 1:
```

File: tutorials/platform/android/android_in_app_purchases.rst [hunk 2]
```diff
@@ -102,15 +102,28 @@ initiate the purchase flow for it.
...
+Then, wait for the ``_on_purchases_updated`` callback and handle the purchase result:
 
 
+::
+
+    func _on_purchases_updated(purchases):
+        for purchase in purchases:
+            if purchase.purchase_state==1: # 1 means "purchased"
+                # enable_premium(purchase.sku) # unlock paid content, add coins, save token on server, etc.
+                if !purchase.is_acknowledged:
```
Comment (timoschwarzer) [inline tutorials/platform/android/android_in_app_purchases.rst:115]:
```suggestion
                if not purchase.is_acknowledged:                                        
```

File: tutorials/platform/android/android_in_app_purchases.rst [hunk 3]
```diff
@@ -102,15 +102,28 @@ initiate the purchase flow for it.
...
     payment.purchase("my_iap_item")
+    
+Then, wait for the ``_on_purchases_updated`` callback and handle the purchase result:
 
 
+::
+
+    func _on_purchases_updated(purchases):
+        for purchase in purchases:
+            if purchase.purchase_state==1: # 1 means "purchased"
```
Comment (timoschwarzer) [inline tutorials/platform/android/android_in_app_purchases.rst:113]:
```suggestion
            if purchase.purchase_state == 1: # 1 means "purchased", see https://developer.android.com/reference/com/android/billingclient/api/Purchase.PurchaseState#constants_1
```

File: tutorials/platform/android/android_in_app_purchases.rst [hunk 4]
```diff
@@ -102,15 +102,28 @@ initiate the purchase flow for it.
...
+                # enable_premium(purchase.sku) # unlock paid content, add coins, save token on server, etc.
```
Comment (timoschwarzer) [inline tutorials/platform/android/android_in_app_purchases.rst:114]:
```suggestion
                # enable_premium(purchase.sku) # unlock paid content, add coins, save token on server, etc. (you have to implement enable_premium yourself)
```

File: tutorials/platform/android/android_in_app_purchases.rst [hunk 5]
```diff
@@ -132,17 +146,18 @@ If your in-app item is not a one-time purchase but a consumable item (e.g. coins
...
 
     var query = payment.queryPurchases("inapp") # Or "subs" for subscriptions
     if query.status == OK:
         for purchase in query.purchases:
-            if purchase.sku == "my_consumable_iap_item":
-                if !purchase.is_acknowledged:
-                    payment.consumePurchase(purchase.purchase_token)
-                    # Check the _on_purchase_consumed callback and give the user what they bought
-
+            if purchase.sku == "my_consumable_iap_item" and purchase.purchase_state==1:
```
Comment (timoschwarzer) [inline tutorials/platform/android/android_in_app_purchases.rst:156]:
```suggestion
            if purchase.sku == "my_consumable_iap_item" and purchase.purchase_state == 1:
```

**Review by timoschwarzer — CHANGES_REQUESTED** [review]
Thank you! Only some small changes :)

File: tutorials/platform/android/android_in_app_purchases.rst [hunk 6]
```diff
@@ -102,15 +102,28 @@ initiate the purchase flow for it.
 ::
 
     payment.purchase("my_iap_item")
+
```
Comment (mhilbrunner) [inline tutorials/platform/android/android_in_app_purchases.rst:105]:
```suggestion

```

File: tutorials/platform/android/android_in_app_purchases.rst [hunk 7]
```diff
@@ -132,17 +146,18 @@ If your in-app item is not a one-time purchase but a consumable item (e.g. coins
...
-            if purchase.sku == "my_consumable_iap_item":
-                if !purchase.is_acknowledged:
-                    payment.consumePurchase(purchase.purchase_token)
-                    # Check the _on_purchase_consumed callback and give the user what they bought
-
+            if purchase.sku == "my_consumable_iap_item" and purchase.purchase_state == 1:
+                # enable_premium(purchase.sku) # add coins, save token on server, etc.
+                payment.consumePurchase(purchase.purchase_token)
+                # Or wait for the _on_purchase_consumed callback before giving the user what they bought
+
```
Comment (mhilbrunner) [inline tutorials/platform/android/android_in_app_purchases.rst:160]:
```suggestion

```

**Review by mhilbrunner — APPROVED** [review]

File: tutorials/platform/android/android_in_app_purchases.rst [hunk 8]
```diff
@@ -102,15 +102,27 @@ initiate the purchase flow for it.
 ::
 
     payment.purchase("my_iap_item")
+Then, wait for the ``_on_purchases_updated`` callback and handle the purchase result:
 
 
+::
```
Comment (mhilbrunner) [inline tutorials/platform/android/android_in_app_purchases.rst:108]:
```suggestion

Then, wait for the ``_on_purchases_updated`` callback and handle the purchase result:

::
```

**Comment by mhilbrunner** [comment]
Merged. Thanks for this, and congrats on your first contribution to the Godot docs! :)

Small note: In future Pull Requests, please use clear commit messages instead of the autogenerated default.

**Comment by avnerh1** [comment]
> Merged. Thanks for this, and congrats on your first contribution to the Godot docs! :)
> 
> Small note: In future Pull Requests, please use clear commit messages instead of the autogenerated default.

Thank you!

**Pull request merged**